### ISSUE TITLE
Update and Delete cannot accurately hit the partition table

### ISSUE THREAD

**Issue by shijunfeng** [timescale/timescaledb#2247]
**Relevant system information:**
 - OS: [e.g. CentOS7, etc]
 - PostgreSQL version (output of `postgres --version`): [e.g. 12.0]
 - TimescaleDB version (output of `\dx` in `psql`): [e.g. 1.7.0]
 - Installation method: [e.g. "source"]

**Describe the bug**
Update and Delete cannot accurately hit the partition table

CREATE TABLE "public"."m_business" (
  "b_code" varchar(21) COLLATE "pg_catalog"."default" NOT NULL,
  "b_times" timestamp(0) NOT NULL,
  "b_tag" varchar(32) COLLATE "pg_catalog"."default" NOT NULL,
  "b_value" varchar(255) COLLATE "pg_catalog"."default" DEFAULT NULL::character varying,
  "b_createtime" timestamp(0) NOT NULL,
  "b_value1" varchar(255) COLLATE "pg_catalog"."default" DEFAULT NULL::character varying,
  "b_value2" varchar(255) COLLATE "pg_catalog"."default" DEFAULT NULL::character varying
)
;

-- ----------------------------
-- Indexes structure for table metadata_business
-- ----------------------------
CREATE INDEX "m_business_b_times_idx1" ON "public"."m_business" USING btree (
  "b_times" "pg_catalog"."timestamp_ops" DESC NULLS FIRST
);
select create_hypertable('m_business', 'b_times', chunk_time_interval => interval '10 day',number_partitions => 30, partitioning_column => 'b_tag', if_not_exists => true);

explain analyze UPDATE m_business set b_value2='111' where bk_timestamp='2020-06-29 16:56:13' and b_tag='51f72926013047bc850fa4a17e443af1'

Update on m_business  (cost=0.00..281.24 rows=38 width=1420) (actual time=1336.982..1336.983 rows=0 loops=1)
  Update on m_business
  Update on _hyper_15_29289_chunk
  Update on _hyper_15_29290_chunk
  Update on _hyper_15_29291_chunk
  Update on _hyper_15_29292_chunk
  Update on _hyper_15_29293_chunk
  Update on _hyper_15_29294_chunk
  Update on _hyper_15_29295_chunk
  Update on _hyper_15_29296_chunk
  Update on _hyper_15_29297_chunk
  Update on _hyper_15_29298_chunk
  Update on _hyper_15_29299_chunk
  Update on _hyper_15_29300_chunk
  Update on _hyper_15_29301_chunk
  Update on _hyper_15_29302_chunk
  Update on _hyper_15_29303_chunk
  Update on _hyper_15_29304_chunk
  Update on _hyper_15_29305_chunk
  Update on _hyper_15_29306_chunk
  Update on _hyper_15_29307_chunk
  Update on _hyper_15_29308_chunk
  Update on _hyper_15_29309_chunk
  Update on _hyper_15_29310_chunk
  Update on _hyper_15_29311_chunk
  Update on _hyper_15_29312_chunk
  Update on _hyper_15_29313_chunk
  Update on _hyper_15_29314_chunk
  Update on _hyper_15_29315_chunk
  Update on _hyper_15_29316_chunk
  Update on _hyper_15_29317_chunk
  Update on _hyper_15_29318_chunk
  ->  Seq Scan on m_business  (cost=0.00..0.00 rows=1 width=1712) (actual time=0.012..0.012 rows=0 loops=1)
        Filter: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29289_29289_m_business_qianyi_pkey1" on _hyper_15_29289_chunk  (cost=0.42..12.45 rows=2 width=1712) (actual time=38.844..38.844 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29290_29290_m_business_qianyi_pkey1" on _hyper_15_29290_chunk  (cost=0.42..8.45 rows=1 width=601) (actual time=25.128..25.128 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29291_29291_m_business_qianyi_pkey1" on _hyper_15_29291_chunk  (cost=0.41..12.44 rows=2 width=1712) (actual time=17.776..17.776 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29292_29292_m_business_qianyi_pkey1" on _hyper_15_29292_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=11.307..11.307 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29293_29293_m_business_qianyi_pkey1" on _hyper_15_29293_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=10.630..10.630 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29294_29294_m_business_qianyi_pkey1" on _hyper_15_29294_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=37.075..37.075 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29295_29295_m_business_qianyi_pkey1" on _hyper_15_29295_chunk  (cost=0.42..12.45 rows=2 width=1712) (actual time=60.443..60.443 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29296_29296_m_business_qianyi_pkey1" on _hyper_15_29296_chunk  (cost=0.42..8.45 rows=1 width=601) (actual time=60.847..60.847 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29297_29297_m_business_qianyi_pkey1" on _hyper_15_29297_chunk  (cost=0.42..8.45 rows=1 width=599) (actual time=6.300..6.300 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using _hyper_15_29298_chunk_m_business_qianyi_b_tag_bk_time on _hyper_15_29298_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=118.673..118.674 rows=0 loops=1)
        Index Cond: (((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text) AND (b_times = '2020-06-29 16:56:13'::timestamp without time zone))
  ->  Index Scan using "29299_29299_m_business_qianyi_pkey1" on _hyper_15_29299_chunk  (cost=0.42..12.45 rows=2 width=1712) (actual time=17.924..17.924 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29300_29300_m_business_qianyi_pkey1" on _hyper_15_29300_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=106.514..106.514 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29301_29301_m_business_qianyi_pkey1" on _hyper_15_29301_chunk  (cost=0.42..12.45 rows=2 width=1712) (actual time=58.674..58.674 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29302_29302_m_business_qianyi_pkey1" on _hyper_15_29302_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=39.365..39.365 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29303_29303_m_business_qianyi_pkey1" on _hyper_15_29303_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=15.018..15.018 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29304_29304_m_business_qianyi_pkey1" on _hyper_15_29304_chunk  (cost=0.43..8.45 rows=1 width=604) (actual time=54.379..54.379 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29305_29305_m_business_qianyi_pkey1" on _hyper_15_29305_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=60.557..60.558 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29306_29306_m_business_qianyi_pkey1" on _hyper_15_29306_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=98.472..98.472 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29307_29307_m_business_qianyi_pkey1" on _hyper_15_29307_chunk  (cost=0.43..8.45 rows=1 width=600) (actual time=57.035..57.035 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29308_29308_m_business_qianyi_pkey1" on _hyper_15_29308_chunk  (cost=0.42..12.45 rows=2 width=1712) (actual time=49.081..49.081 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29309_29309_m_business_qianyi_pkey1" on _hyper_15_29309_chunk  (cost=0.41..12.44 rows=2 width=1712) (actual time=53.408..53.408 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29310_29310_m_business_qianyi_pkey1" on _hyper_15_29310_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=31.259..31.259 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29311_29311_m_business_qianyi_pkey1" on _hyper_15_29311_chunk  (cost=0.43..8.45 rows=1 width=605) (actual time=17.275..17.275 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29312_29312_m_business_qianyi_pkey1" on _hyper_15_29312_chunk  (cost=0.43..8.45 rows=1 width=601) (actual time=33.204..33.204 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using _hyper_15_29313_chunk_m_business_qianyi_b_tag_bk_time on _hyper_15_29313_chunk  (cost=0.42..8.45 rows=1 width=602) (actual time=12.919..12.919 rows=0 loops=1)
        Index Cond: (((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text) AND (b_times = '2020-06-29 16:56:13'::timestamp without time zone))
  ->  Index Scan using "29314_29314_m_business_qianyi_pkey1" on _hyper_15_29314_chunk  (cost=0.42..8.45 rows=1 width=601) (actual time=13.905..13.905 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29315_29315_m_business_qianyi_pkey1" on _hyper_15_29315_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=28.580..28.580 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29316_29316_m_business_qianyi_pkey1" on _hyper_15_29316_chunk  (cost=0.42..8.45 rows=1 width=601) (actual time=14.786..14.786 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29317_29317_m_business_qianyi_pkey1" on _hyper_15_29317_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=37.540..37.541 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
  ->  Index Scan using "29318_29318_m_business_qianyi_pkey1" on _hyper_15_29318_chunk  (cost=0.41..8.43 rows=1 width=1712) (actual time=150.012..150.012 rows=0 loops=1)
        Index Cond: ((b_times = '2020-06-29 16:56:13'::timestamp without time zone) AND ((b_tag)::text = '51f72926013047bc850fa4a17e443af1'::text))
Planning Time: 6250.042 ms
Execution Time: 1338.014 ms

**Comment by mfreed** [comment]
Hi @shijunfeng I'm not sure I understand your bug report that "Update and Delete cannot accurately hit the partition table"?

Are you referring to the fact that UPDATE/DELETE don't perform constraint exclusion to exclude certain chunks of the hypertable, and instead execute against each chunk (unlike SELECT & INSERT)?

If so, yes, that is currently a known limitation that effects UPDATE/DELETE performance.

**Comment by shijunfeng** [comment]
Yes, that's the question。Is there any plan to repair it in the near future。In addition, when will the multi node version be officially released？

**Comment by shijunfeng** [comment]
> 是的，这就是问题所在。在不久的将来，是否有任何计划对其进行修复。此外，多节点版本何时正式发布？



> Hi @shijunfeng I'm not sure I understand your bug report that "Update and Delete cannot accurately hit the partition table"?
> 
> Are you referring to the fact that UPDATE/DELETE don't perform constraint exclusion to exclude certain chunks of the hypertable, and instead execute against each chunk (unlike SELECT & INSERT)?
> 
> If so, yes, that is currently a known limitation that effects UPDATE/DELETE performance.

Yes, that's the question。Is there any plan to repair it in the near future。In addition, when will the multi node version be officially released？

**Comment by shijunfeng** [comment]
> Hi @shijunfeng I'm not sure I understand your bug report that "Update and Delete cannot accurately hit the partition table"?
> 
> Are you referring to the fact that UPDATE/DELETE don't perform constraint exclusion to exclude certain chunks of the hypertable, and instead execute against each chunk (unlike SELECT & INSERT)?
> 
> If so, yes, that is currently a known limitation that effects UPDATE/DELETE performance.

Yes, that's the question。Is there any plan to repair it in the near future。In addition, when will the multi node version be officially released？

**Comment by nikkhils** [comment]
> Yes, that's the question。Is there any plan to repair it in the near future。In addition, when will the multi node version be officially released？

@shijunfeng multinode initial version has been released. Please try it out.

**Comment by NunoFilipeSantos** [comment]
@shijunfeng please let us know if the new release works for you.
